File: app/src/main/java/home/oleg/placesnearme/app/di/components/ApplicationComponent.java
package home.oleg.placesnearme.app.di.components;

import javax.inject.Singleton;

import dagger.Component;
import home.oleg.placesnearme.api.AppApi;
import home.oleg.placesnearme.app.PlacesNearMeApp;
import home.oleg.placesnearme.app.di.modules.CoreModule;
import home.oleg.placesnearme.app.di.modules.DbModule;
import home.oleg.placesnearme.app.di.modules.InteractorModule;
import home.oleg.placesnearme.app.di.modules.NetworkModule;
import home.oleg.placesnearme.app.di.modules.RepositoryModule;
import home.oleg.placesnearme.app.di.modules.ViewModelModule;
import home.oleg.placesnearme.builder.ComponentBuilder;

@Component(modules = {
        CoreModule.class,
        DbModule.class,
        NetworkModule.class,
        RepositoryModule.class,
        InteractorModule.class,
        ViewModelModule.class})
@Singleton
public interface ApplicationComponent extends AppApi {

    @Component.Builder
    interface Builder extends ComponentBuilder<ApplicationComponent, PlacesNearMeApp> {

    }

}

File: app/src/main/java/home/oleg/placesnearme/app/di/components/ApplicationComponent.kt
package home.oleg.placesnearme.app.di.components

import dagger.Component
import home.oleg.placesnearme.app.PlacesNearMeApp
import home.oleg.placesnearme.app.di.modules.CoreModule
import home.oleg.placesnearme.coredi.api.AppApi
import home.oleg.placesnearme.coredi.builder.ComponentBuilder
import javax.inject.Singleton

@Component(modules = [CoreModule::class])
@Singleton
interface ApplicationComponent : AppApi {

    @Component.Builder
    interface Builder : ComponentBuilder<ApplicationComponent, PlacesNearMeApp>

}
